assets/js/services/buySell.service.js

```diff
@@ -176,17 +176,16 @@ function buySell ($rootScope, $timeout, $q, $uibModal, Wallet, MyWallet, MyWalle
       .then(lean ? () => {} : success, error);
   }
 
-  function openBuyView (transaction, trade, bitcoinReceived) {
+  function openBuyView (trade = null, options = {}) {
     return $uibModal.open({
       templateUrl: 'partials/buy-modal.jade',
       windowClass: 'bc-modal auto buy',
       controller: 'BuyCtrl',
       backdrop: 'static',
       keyboard: false,
       resolve: {
-        bitcoinReceived: () => bitcoinReceived || undefined,
-        trade: () => trade || null,
-        transaction: () => transaction || {}
+        trade: () => trade && trade.refresh(),
```

Comment:
trade.refresh() returns a promise that resolves with the trade, so it shouldn't set trade to undefined.